src/lib/dialog/dialog-container.ts

```diff
@@ -182,8 +182,13 @@ export class MdDialogContainer extends BasePortalHost {
       this._restoreFocus();
     }
 
-    this._animationStateChanged.emit(event);
-    this._isAnimating = false;
+    // Note: as of Angular 4.3, the animations module seems to fire the `start` callback before
+    // the end if animations are disabled. Make this call async to ensure that it still fires
+    // at the appropriate time.
+    Promise.resolve().then(() => {
```

Comment:
Add a test for this fix? The current tests are using the `NoopAnimationsModule` and they didn't catch the issue.

Reply:
I looked into a way to reproduce that issue inside of the `TestBed`, but it looks like the animation callbacks fire correctly there.

I tried to look into the `NoopAnimationPlayer` from `@angular/animations` and I assume its related to some timing issues with zone.js